Conformance check

xcomagent95 opened the pull request:
The does now check the API for its conformance. If a possible non-compliance could be identified it is noted in the help texts of the tool. The "No information provided" messages are now also removed. The server url is now also passed to the R-script via CDATA. The nullable=true and minOccurs=0 also lead to a optional parameter thus fixing issues #5 #13 #9 #12.

xcomagent95 committed:
First simple conformance test

xcomagent95 committed:
Conformance cheack and some comments

MarkusKonk committed:
change description and help tags

MarkusKonk committed:
several changes

bgruening commented on generic.xml:
```diff
@@ -15,174 +15,2263 @@
 	</outputs>
 	<expand macro="tests"/>
-	<expand macro="help"/>
+	<help>This tool is a wrapper for OGC API Processes (OTB) coming from the Zoo Project (https://zoo-project.github.io/docs/intro.html) and was created using the OGC-API-Process2Galaxy tool (https://github.com/AquaINFRA/OGC-API-Process2Galaxy). Check the README in the repository for more information.</help>
```
This should not go into a tool help, please use the `<creator>` tag. The tool help should assist a user in using the tool.

xcomagent95 committed:
Changes to properly handle image inputs

MarkusKonk committed:
minor changes to config

MarkusKonk committed:
oneOf issue

MarkusKonk committed:
new xml

xcomagent95 committed:
Fixed issues regarding the server parameter, help
messages and optional parameters.
The server url is now passed via the CDATA to the
R-script. The "Nor information provided" messages
are removed and the minOccurs and nullable parameter
values now lead to an optional parameter this
fixing issues #13, #12 and #9
